### ISSUE TITLE
Modeller Running and Persistence

### ISSUE THREAD

**Issue by salaboy** [Activiti/Activiti#2432]
Enable a single docker image that can store files to the file system on save/export
@daisuke-yoshimoto already started with merging BE and FE into a single docker image and security configurations with basic auth. This will extend this work with FS persistence implementation of the modelling layers

@daisuke-yoshimoto can you please link your issue here.

**Comment by daisuke-yoshimoto** [comment]
@salaboy 
Is it okay if I am in charge of this issue too?

**Comment by salaboy** [comment]
@daisuke-yoshimoto that will be great.. we have quite tight deadlines.. 
We need to get a new implementation of the modeling APIs on the FileSystem.. do you know where to start that?
We can ask @constantin-ciobotaru to give you some directions to understand which implementations to provide first.

**Comment by salaboy** [comment]
@constantin-ciobotaru if we want to store projects (for example) in the FileSystem should we provide another implementation for this interface right -> https://github.com/Activiti/activiti-cloud-org-service/blob/develop/activiti-cloud-services-org/activiti-cloud-services-org-repository/src/main/java/org/activiti/cloud/organization/repository/ProjectRepository.java#L28 ?

**Comment by constantin-ciobotaru** [comment]
@salaboy @daisuke-yoshimoto Yes, we should provide implementations of [ProjectRepository](https://github.com/Activiti/activiti-cloud-org-service/blob/develop/activiti-cloud-services-org/activiti-cloud-services-org-repository/src/main/java/org/activiti/cloud/organization/repository/ProjectRepository.java) and [ModelRepository](https://github.com/Activiti/activiti-cloud-org-service/blob/develop/activiti-cloud-services-org/activiti-cloud-services-org-repository/src/main/java/org/activiti/cloud/organization/repository/ModelRepository.java) interfaces that stores the projects as filesystem folders and models as files in that folder. [ProjectImpl](https://github.com/Activiti/activiti-cloud-org-service/blob/develop/activiti-cloud-services-org/activiti-cloud-services-org-api-impl/src/main/java/org/activiti/cloud/organization/api/impl/ProjectImpl.java) and [ModelImpl](https://github.com/Activiti/activiti-cloud-org-service/blob/develop/activiti-cloud-services-org/activiti-cloud-services-org-api-impl/src/main/java/org/activiti/cloud/organization/api/impl/ModelImpl.java) can be used as entities. [ObjectMapperConfiguration](https://github.com/Activiti/activiti-cloud-org-service/blob/develop/activiti-cloud-services-org/activiti-cloud-services-org-api-impl/src/main/java/org/activiti/cloud/organization/config/ObjectMapperConfiguration.java) is the configuration that maps those implementations to interfaces and should be loaded in context as bean. I hope this should be enough, but if not, I can help with any issue that will appear.

**Comment by salaboy** [comment]
@constantin-ciobotaru can you point us also to the interfaces in process-model-service as well? 
Also.. which maven modules with the new implementation should we create?

**Comment by constantin-ciobotaru** [comment]
@salaboy process-model-service doesn't implement any interface, we did this "magic" only in organization. About the org module probably the activiti-cloud-services-org-filesystem (sibling with activiti-cloud-services-org-jpa) should be the module to contain the new repository implementations (something like FilesystemProjectRepository and FilesystemModelRepository)

**Comment by salaboy** [comment]
@constantin-ciobotaru great.. so @daisuke-yoshimoto the requirement is clear now for me.. if you want to get started with that after finishing the other issue that will be great..

**Comment by constantin-ciobotaru** [comment]
@salaboy @daisuke-yoshimoto The switch between jpa and filesystem should be done with an application property and @ConditionalOnProperty on the beans? Eventually, the property can be "spring.datasource.url" with filesystem implementation as default.

**Comment by daisuke-yoshimoto** [comment]
@salaboy 
Yes. I have the same opinion.
Modeler is simply an editor for bpmn and json, it does not provide interaction with Git.

**Comment by daisuke-yoshimoto** [comment]
@salaboy 
Also, is this a special feature of the standalone version?
Or can we use it other than standalone version?

**Comment by constantin-ciobotaru** [comment]
@salaboy @daisuke-yoshimoto If we still want to offer a model versioning mechanism, I think it worth to think at an interaction with git (using maybe jgit). For now we don't have endpoints for model versions, but if we plan this, the filesystem based implementation will need such a mechanism. And git is a good choice, IMO.

**Comment by daisuke-yoshimoto** [comment]
@salaboy @constantin-ciobotaru 
For now, I will start implementing with the feeling of outputting only the latest version to a file.
Can I put the implementation in activiti-cloud-modeling-standalone repo? Or activiti-cloud-org-service repo?

**Comment by salaboy** [comment]
@daisuke-yoshimoto it should go to the org-service repo.. so we keep it as a separate implementation that we can choose from.

**Comment by daisuke-yoshimoto** [comment]
@salaboy 
Ok. Do you mean that the file system mode also supports cluster environments?
In that case, I need to make it properly in consideration of exclusive locking mechanism.

**Comment by salaboy** [comment]
@daisuke-yoshimoto not really.. let's not create a locking mechanism. If the POD writes in a cluster (cloud) it will be probably in a local volume, which shouldn't be accessed by other services. For now let's keep it simple. we can always iterate.
